shell/src/test/java/alluxio/cli/bundler/command/CollectLogCommandTest.java

```diff
@@ -19,56 +22,500 @@
+  public void fileNameIncluded() throws Exception {
+    InfoCollectorTestUtils.createFileInDir(mTestDir, "alluxio_gc.log");
+    InfoCollectorTestUtils.createFileInDir(mTestDir, "alluxio_gc.log.1");
+    InfoCollectorTestUtils.createFileInDir(mTestDir, "alluxio_gc.log.2");
 
+    CollectLogCommand cmd = new CollectLogCommand(FileSystemContext.create(mConf));
     File targetDir = InfoCollectorTestUtils.createTemporaryDirectory();
     CommandLine mockCommandLine = mock(CommandLine.class);
-    String[] mockArgs = new String[]{targetDir.getAbsolutePath()};
+    String[] mockArgs = new String[]{
+            cmd.getCommandName(),
+            targetDir.getAbsolutePath()
+    };
     when(mockCommandLine.getArgs()).thenReturn(mockArgs);
+    when(mockCommandLine.hasOption(eq("include-logs"))).thenReturn(true);
+    when(mockCommandLine.getOptionValue(eq("include-logs"))).thenReturn("alluxio_gc");
     int ret = cmd.run(mockCommandLine);
-    Assert.assertEquals(0, ret);
+    assertEquals(0, ret);
 
     // Files will be copied to sub-dir of target dir
-    File subDir = new File(Paths.get(targetDir.getAbsolutePath(), cmd.getCommandName()).toString());
-
-    // Check the dir copied
-    String[] files = subDir.list();
-    Arrays.sort(files);
-    String[] expectedFiles = sTestDir.list();
-    Arrays.sort(expectedFiles);
-    Assert.assertEquals(expectedFiles, files);
+    File subDir = new File(targetDir, cmd.getCommandName());
+    mExpectedFiles.add("alluxio_gc.log");
+    mExpectedFiles.add("alluxio_gc.log.1");
+    mExpectedFiles.add("alluxio_gc.log.2");
+
+    InfoCollectorTestUtils.verifyAllFiles(subDir, mExpectedFiles);
+  }
+
+  @Test
+  public void endTimeFilter() throws Exception {
+    // Define an issue end datetime
+    // We ignore logs that are created after this
+    LocalDateTime issueEnd = LocalDateTime.of(2020, 7, 8, 18, 0, 0);
+    DateTimeFormatter fmt = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    // Other logs all end before this issue
+    LocalDateTime logEnd = issueEnd.minusHours(1);
+    long logEndTimestamp = logEnd.toEpochSecond(ZoneOffset.UTC);
+    for (File f : CommonUtils.recursiveListDir(mTestDir)) {
+      f.setLastModified(logEndTimestamp);
+    }
+
+    // Some logs are created after this time, should be ignored
+    File masterLog = new File(mTestDir, "master.log");
+    String log = "2020-07-08 18:53:45,129 INFO  CopycatServer - Server started successfully!\n"
+            + "2020-07-08 18:53:59,129 INFO  RaftJournalSystem - Started Raft Journal System..\n"
+            + "2020-07-09 00:01:59,135 INFO  DefaultMetaMaster - Standby master with address...\n"
+            + "2020-07-09 00:03:59,135 INFO  AlluxioMasterProcess - All masters started\n"
+            + "2020-07-09 01:12:59,138 INFO  AbstractPrimarySelector - Primary selector..\n"
+            + "2020-07-09 01:53:59,139 INFO  AbstractMaster - TableMaster: Stopped secondary..";
```

Comment:
It might be controversial whether we use real content or abstractions here. I would argue real logs capture more.